lib/slavable.rb

```diff
@@ -18,8 +56,23 @@ def #{with_name}(*args, &block)
         end
       eoruby
 
-      alias_method without_name, method
-      alias_method method, with_name
+      if singleton_class?
+        begin
+          alias_method without_name, method
+          alias_method method, with_name
+        rescue NameError
+          @switched_methods ||= []
+          @switched_methods << method
+        end
+      else
+        if (instance_methods.include?(method) || methods.include?(method))
```

Comment:
Convert if nested inside else to elsif.<br>Don't use parentheses around the condition of an if.